service/src/main/java/org/mskcc/cmo/metadb/service/impl/MetaDbRequestServiceImpl.java

```diff
@@ -45,17 +58,33 @@ public void saveRequest(MetaDbRequest request) throws Exception {
                 request.setMetaDbSampleList(updatedSamples);
             }
             metaDbRequestRepository.save(request);
+            return true;
         } else {
-            for (MetaDbSample s: request.getMetaDbSampleList()) {
-                if (s.getSampleIgoId() != null
-                        && metaDbRequestRepository.findMetaDbSampleByRequestAndIgoId(request.getRequestId(),
-                        s.getSampleIgoId().getSampleId()) == null) {
-                    savedRequest.addMetaDbSampleList(s);
-                    metaDbRequestRepository.save(savedRequest);
-                }
-            }
+            
```

Comment:
I think in general it's best to not create and use something in the same line that has the potential to throw an exception

Separate the declaration of the File returned by `fileUtil.getOrCreateFileWithHeader()`  and then use the returned object in `fileUtil.writeToFile()`